modules/recon/hosts-ports/binaryedge_ports.py

```diff
@@ -0,0 +1,23 @@
+from recon.core.module import BaseModule
+
+class Module(BaseModule):
+
+    meta = {
+        'name': 'BinaryEdge.io Ports lookup',
+        'author': 'Ryan Hays',
+        'version': '1.0',
+        'description': 'Uses the BinaryEdge API to discover open services for IP Addresses.',
+        'required_keys': ['binaryedge_api'],
+        'query': 'SELECT DISTINCT ip_address FROM hosts WHERE ip_address IS NOT NULL',
+    }
+
+    def module_run(self, ipaddrs):
+        key = self.get_key('binaryedge_api')
+        for ipaddr in ipaddrs:
+            self.heading(ipaddr, level=0)
+            resp = self.request(url=f"https://api.binaryedge.io/v2/query/ip/{ipaddr}", headers={'X-Key': key})
+            if resp.status_code == 200:
+                for event in resp.json().get('events'):
+                    for result in event['results']:
+                        self.insert_ports(ip_address=ipaddr, port=result['target']['port'],
+                                          protocol=result['target']['protocol'])
```

Comment:
Looks good. Please add a newline to the end of the file and update the pull request. Thanks!

Reply:
Should be updated with the newline now.